### ISSUE TITLE
`reportAny` false negative on attribute

### ISSUE THREAD

**Issue by beauxq** [DetachHead/basedpyright#1647]
### Description

`reportAny` isn't working on a class attribute

```py
import json
from typing import reveal_type

class C:
    x: int

    def __init__(self):
        self.x = json.loads("abc")  # expected reportAny
        reveal_type(self.x)

        y: int = json.loads("abc")  # reportAny works here
        reveal_type(y)
```
https://basedpyright.com/?typeCheckingMode=all&enableExperimentalFeatures=true&code=JYWwDg9gTgLgBAKwM4QHYCgBmUIjjATzGFQHM5RJY4oBTAN1oEMAbAfULFvXQGMWmSJHADCALnRwpcAB5iKqGD2lwAJrUxw2bEsBjaAFElotMASgkqVx0wDoZcALyIUqWywhNVSAwCImAEa8vmZSAMRwtDJcvPB0VDAAgqgEklZSdIysHES0RiaY9mbK6XAE8iTwzsho7p7efoHBoXAR8dBJKXAA7tAA1sIAFrR0aemZzOyceQTFQA
